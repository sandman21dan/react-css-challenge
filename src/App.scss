.app {

  @media (min-width: 764px) {
    &__layout {
      display: flex;
      flex-direction: row-reverse;

      & > *:last-child {
        width: 8rem;
      }
    }
  }
}
