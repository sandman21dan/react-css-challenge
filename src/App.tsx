import * as React from 'react';
import './App.css';
import Hello from './components/Hello';
import HasInitialData from './components/HasInitialData';

import logo from './logo.svg';
import { fetchInitialData, IFetchInitialData } from './store/actions';
import { Dispatch } from 'redux';
import { connect } from 'react-redux';

interface IProps {
  onInit?: () => void;
}

export class App extends React.Component<IProps> {
  public componentDidMount() {
    if (this.props.onInit) {
      this.props.onInit();
    }
  }

  public render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.tsx</code> and save to reload.
        </p>

        <Hello/>
        <HasInitialData/>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch: Dispatch<IFetchInitialData>) => ({
  onInit: () => dispatch(fetchInitialData()),
});

export default connect(undefined, mapDispatchToProps)(App as any);
