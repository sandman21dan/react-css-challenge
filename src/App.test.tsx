import * as React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders without crashing', () => {
  const wrapper = shallow(<App/>);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('calls function on init', () => {
  const testFn = jest.fn();
  shallow(<App onInit={testFn}/>);

  expect(testFn).toHaveBeenCalled();
});
