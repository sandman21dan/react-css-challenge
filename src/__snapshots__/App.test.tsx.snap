// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.tsx
    </code>
     and save to reload.
  </p>
  <Connect(Component) />
  <Connect(Component) />
</div>
`;
